<!DOCTYPE html> <html lang="en"> <head> <meta charset="utf-8"> <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"> <meta http-equiv="X-UA-Compatible" content="IE=edge"> <title>publications | Mihir Bafna</title> <meta name="author" content="Mihir Bafna"/> <meta name="description" content=""/> <link href="https://cdn.jsdelivr.net/npm/bootstrap@4.6.1/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha256-DF7Zhf293AJxJNTmh5zhoYYIMs2oXitRfBjY+9L//AY=" crossorigin="anonymous"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/css/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous"/> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5.15.4/css/all.min.css" integrity="sha256-mUZM63G8m73Mcidfrv5E+Y61y7a12O5mW4ezU3bxqW4=" crossorigin="anonymous"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/academicons@1.9.1/css/academicons.min.css" integrity="sha256-i1+4qU2G2860dGGIOJscdC30s9beBXjFfzjWLjBRsBg=" crossorigin="anonymous"> <link rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Roboto+Slab:100,300,400,500,700|Material+Icons"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/jwarby/jekyll-pygments-themes@master/github.css" media="none" id="highlight_theme_light"/> <link rel="shortcut icon" href="/assets/img/dna.png"/> <link rel="stylesheet" href="/assets/css/main.css"> <link rel="canonical" href="https://mihirbafna.github.io/publications/"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/jwarby/jekyll-pygments-themes@master/native.css" media="none" id="highlight_theme_dark"/> <script src="/assets/js/theme.js"></script> <script src="/assets/js/dark_mode.js"></script> </head> <body class="fixed-top-nav "> <header> <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top"> <div class="container"> <a class="navbar-brand title font-weight-lighter" href="https://mihirbafna.github.io/"><span class="font-weight-bold">Mihir</span> Bafna</a> <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar top-bar"></span> <span class="icon-bar middle-bar"></span> <span class="icon-bar bottom-bar"></span> </button> <div class="collapse navbar-collapse text-right" id="navbarNav"> <ul class="navbar-nav ml-auto flex-nowrap"> <link rel="stylesheet" href="../assets/css/dna.css"> <li class="nav-item "> <a class="nav-link" href="/">about</a> </li> <li class="nav-item active"> <a class="nav-link" href="/publications/">publications<span class="sr-only">(current)</span></a> </li> <li class="nav-item "> <a class="nav-link" href="/experience/">experience</a> </li> <li class="nav-item "> <a class="nav-link" href="/cv/">cv</a> </li> <div class="toggle-container"> <a id="light-toggle"> <i class="fas fa-sun"></i> <i class="fas fa-moon"></i> </a> </div> </ul> </div> </div> </nav> </header> <div class="container mt-5"> <div class="post"> <header class="post-header"> <h1 class="post-title">publications</h1> <p class="post-description"></p> </header> <article> <div class="publications"> <div class="pub-header"> <div class="pub-source">Periodically updated from <a class="pub-source-link" href="https://scholar.google.com/citations?user=wZLz918AAAAJ" target="_blank" rel="noopener noreferrer">google scholar</a>.</div> <div class="pub-note">(†=equal contribution)</div> </div> <h2 class="year">2025</h2> <ol class="bibliography"> <li> <div class="row"> <div class="col-sm-2 abbr"> <abbr class="badge">MLSB</abbr><br> <div class="badge award" style="box-shadow: none;">Oral</div> </div> <div id="jingbafna2025switchcraft" class="col-sm-9"> <div class="title">SwitchCraft: Programmatic Design of State-Switching Proteins</div> <div class="author"> <a href="https://people.csail.mit.edu/bjing/" target="_blank" rel="noopener noreferrer">Bowen Jing<sup>†</sup></a>,  <em>Mihir Bafna<sup>†</sup></em>, Adam Klivans, and <a href="https://people.csail.mit.edu/bab/" target="_blank" rel="noopener noreferrer">Bonnie Berger</a> </div> <div class="periodical"> Machine Learning for Structural Biology Dec 2025 </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> </div> <div class="abstract hidden"> <p>Complex multistate functional mechanisms are observed ubiquitously in natural proteins, yet the path towards systematic de novo rational design of such mechanisms remains unclear, despite significant advancements in protein language models and structure diffusion models. We introduce SwitchCraft, a versatile and programmatic framework for state-switching proteins based on backpropagation through compositional design constraints parameterized by structure prediction models. Our in silico evaluations demonstrate success on a wide range of state-switching design specifications, from allosteric regulation of functional motifs to discrimination of bound ligand identities. Notably, one design exhibits a 3.8 A conformational change upon oxygenation of heme, mimicking mechanisms of cooperativity in hemoglobin. These results position SwitchCraft at the inception of a powerful paradigm for higher-order functional protein design.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"> <abbr class="badge">biorxiv</abbr> </div> <div id="jing2025generating" class="col-sm-9"> <div class="title">Generating functional and multistate proteins with a multimodal diffusion transformer</div> <div class="author"> <a href="https://people.csail.mit.edu/bjing/" target="_blank" rel="noopener noreferrer">Bowen Jing<sup>†</sup></a>, Anna Sappington<sup>†</sup>,  <em>Mihir Bafna</em>, Ravi Shah, Adrina Tang, Rohith Krishna, Adam Klivans, Daniel J Diaz, and <a href="https://people.csail.mit.edu/bab/" target="_blank" rel="noopener noreferrer">Bonnie Berger</a> </div> <div class="periodical"> bioRxiv Sep 2025 </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://www.biorxiv.org/content/10.1101/2025.09.03.672144v2" class="btn btn-sm z-depth-0" role="button" target="_blank" rel="noopener noreferrer">bioRXiv</a> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> </div> <div class="abstract hidden"> <p>Generating proteins with the full diversity and complexity of functions found in nature is a grand challenge in protein design. Here, we present ProDiT, a multimodal diffusion model that unifies sequence and structure modeling paradigms to enable the design of functional proteins at scale. Trained on sequences, 3D structures, and annotations for 214M proteins across the evolutionary landscape, ProDiT generates diverse, novel proteins that preserve known active and binding site motifs and can be successfully conditioned on a wide range of molecular functions, spanning 465 Gene Ontology terms. We introduce a diffusion sampling protocol to design proteins with multiple functional states, and demonstrate this protocol by scaffolding active sites from enzymes such as carbonic anhydrase and lysozyme to be allosterically deactivated by a calcium effector. Our results showcase ProDiT9s unique capacity to satisfy design specifications inaccessible to existing generative models, thereby expanding the protein design toolkit.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">jing2025generating</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Generating functional and multistate proteins with a multimodal diffusion transformer}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Jing, Bowen and Sappington, Anna and Bafna, Mihir and Shah, Ravi and Tang, Adrina and Krishna, Rohith and Klivans, Adam and Diaz, Daniel J and Berger, Bonnie}</span><span class="p">,</span>
  <span class="na">equal</span> <span class="p">=</span> <span class="s">{Jing, Bowen;  Sappington, Anna}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{bioRxiv}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2025}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">sep</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">{Cold Spring Harbor Laboratory}</span><span class="p">,</span>
  <span class="na">short</span> <span class="p">=</span> <span class="s">{biorxiv}</span><span class="p">,</span>
  <span class="na">shorttitle</span> <span class="p">=</span> <span class="s">{biorxiv}</span><span class="p">,</span>
  <span class="na">biorxiv</span> <span class="p">=</span> <span class="s">{10.1101/2025.09.03.672144v2}</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"> <abbr class="badge">arxiv</abbr> </div> <div id="bafna2025learningresiduelevelprotein" class="col-sm-9"> <div class="title">Learning residue level protein dynamics with multiscale Gaussians</div> <div class="author"> <em>Mihir Bafna</em>, <a href="https://people.csail.mit.edu/bjing/" target="_blank" rel="noopener noreferrer">Bowen Jing</a>, and <a href="https://people.csail.mit.edu/bab/" target="_blank" rel="noopener noreferrer">Bonnie Berger</a> </div> <div class="periodical"> arxiv Sep 2025 </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="http://arxiv.org/abs/2509.01038" class="btn btn-sm z-depth-0" role="button" target="_blank" rel="noopener noreferrer">arXiv</a> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> </div> <div class="abstract hidden"> <p>Many methods have been developed to predict static protein structures, however understanding the dynamics of protein structure is essential for elucidating biological function. While molecular dynamics (MD) simulations remain the in silico gold standard, its high computational cost limits scalability. We present DynaProt, a lightweight, SE(3)-invariant framework that predicts rich descriptors of protein dynamics directly from static structures. By casting the problem through the lens of multivariate Gaussians, DynaProt estimates dynamics at two complementary scales: (1) per-residue marginal anisotropy as 3×3 covariance matrices capturing local flexibility, and (2) joint scalar covariances encoding pairwise dynamic coupling across residues. From these dynamics outputs, DynaProt achieves high accuracy in predicting residue-level flexibility (RMSF) and, remarkably, enables reasonable reconstruction of the full covariance matrix for fast ensemble generation. Notably, it does so using orders of magnitude fewer parameters than prior methods. Our results highlight the potential of direct protein dynamics prediction as a scalable alternative to existing methods.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">bafna2025learningresiduelevelprotein</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Learning residue level protein dynamics with multiscale Gaussians}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Bafna, Mihir and Jing, Bowen and Berger, Bonnie}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">sep</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2025}</span><span class="p">,</span>
  <span class="na">eprint</span> <span class="p">=</span> <span class="s">{2509.01038}</span><span class="p">,</span>
  <span class="na">archiveprefix</span> <span class="p">=</span> <span class="s">{arXiv}</span><span class="p">,</span>
  <span class="na">primaryclass</span> <span class="p">=</span> <span class="s">{q-bio.BM}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"> <abbr class="badge">PNAS</abbr> </div> <div id="gujral2025sparse" class="col-sm-9"> <div class="title">Sparse autoencoders uncover biologically interpretable features in protein language model representations</div> <div class="author"> <a href="https://www.linkedin.com/in/onkar-singh-gujral-b9ab8626a/" target="_blank" rel="noopener noreferrer">Onkar Gujral</a>,  <em>Mihir Bafna</em>, <a href="https://be.mit.edu/faculty/eric-alm/" target="_blank" rel="noopener noreferrer">Eric Alm</a>, and <a href="https://people.csail.mit.edu/bab/" target="_blank" rel="noopener noreferrer">Bonnie Berger</a> </div> <div class="periodical"> Proceedings of the National Academy of Sciences Sep 2025 </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://www.pnas.org/doi/10.1073/pnas.2506316122" class="btn btn-sm z-depth-0" role="button" target="_blank" rel="noopener noreferrer">Paper</a> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> </div> <div class="abstract hidden"> <p>Foundation models in biology—particularly protein language models (PLMs)—have enabled ground-breaking predictions in protein structure, function, and beyond. However, the “black-box” nature of these representations limits transparency and explainability, posing challenges for human–AI collaboration and leaving open questions about their human-interpretable features. Here, we leverage sparse autoencoders (SAEs) and a variant, transcoders, from natural language processing to extract, in a completely unsupervised fashion, interpretable sparse features present in both protein-level and amino acid (AA)-level representations from ESM2, a popular PLM. Unlike other approaches such as training probes for features, the extraction of features by the SAE is performed without any supervision. We find that many sparse features extracted from SAEs trained on protein-level representations are tightly associated with Gene Ontology (GO) terms across all levels of the GO hierarchy. We also use Anthropic’s Claude to automate the interpretation of sparse features for both protein-level and AA-level representations and find that many of these features correspond to specific protein families and functions such as the NAD Kinase, IUNH, and the PTH family, as well as proteins involved in methyltransferase activity and in olfactory and gustatory sensory perception. We show that sparse features are more interpretable than ESM2 neurons across all our trained SAEs and transcoders. These findings demonstrate that SAEs offer a promising unsupervised approach for disentangling biologically relevant information present in PLM representations, thus aiding interpretability. This work opens the door to safety, trust, and explainability of PLMs and their applications, and paves the way to extracting meaningful biological insights across increasingly powerful models in the life sciences.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">gujral2025sparse</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Gujral, Onkar and Bafna, Mihir and Alm, Eric and Berger, Bonnie}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{Proceedings of the National Academy of Sciences}</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">{122}</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">{34}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{e2506316122}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2025}</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">{National Academy of Sciences}</span><span class="p">,</span>
  <span class="na">short</span> <span class="p">=</span> <span class="s">{PNAS}</span><span class="p">,</span>
  <span class="na">shorttitle</span> <span class="p">=</span> <span class="s">{PNAS}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"> <abbr class="badge">biorxiv</abbr> </div> <div id="Zhang2020.11.26.400739" class="col-sm-9"> <div class="title">DANGO: Predicting Higher-Order Genetic Interactions</div> <div class="author"> <a href="https://ruochiz.com/" target="_blank" rel="noopener noreferrer">Ruochi Zhang</a>,  <em>Mihir Bafna</em>, <a href="https://majianzhu.com/" target="_blank" rel="noopener noreferrer">Jianzhu Ma</a>, and <a href="https://www.cs.cmu.edu/~jianma/" target="_blank" rel="noopener noreferrer">Jian Ma</a> </div> <div class="periodical"> biorxiv Jan 2025 </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://www.biorxiv.org/content/10.1101/2020.11.26.400739v2" class="btn btn-sm z-depth-0" role="button" target="_blank" rel="noopener noreferrer">bioRXiv</a> <a href="https://www.biorxiv.org/content/early/2025/01/13/2020.11.26.400739.full.pdf" class="btn btn-sm z-depth-0" role="button" target="_blank" rel="noopener noreferrer">Paper</a> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> </div> <div class="abstract hidden"> <p>Higher-order genetic interactions, which have profound implications for understanding the molecular mechanisms of phenotypic variation, remain poorly characterized. Most studies to date have focused on pairwise interactions, as designing high-throughput experimental screenings for the vast combinatorial search space of higher-order molecular interactions is dauntingly challenging. Here, we develop DANGO, a computational method based on a self-attention hypergraph neural network, designed to effectively predict higher-order genetic interaction among groups of genes. As a proof-of-concept, we provide comprehensive predictions for over 400 million trigenic interactions in the yeast S. cerevisiae, significantly expanding the quantitative characterization of such interactions. Our results demonstrate that DANGO accurately predicts trigenic interactions, uncovering both known and novel biological functions related to cell growth. We further incorporate protein embeddings and model uncertainty scoring to enhance the biological relevance and interpretability of the predicted interactions. Moreover, the predicted interactions can serve as powerful genetic markers for growth response under diverse conditions. Together, DANGO enables a more complete map of complex genetic interactions that impinge upon phenotypic diversity.Competing Interest StatementThe authors have declared no competing interest.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">Zhang2020.11.26.400739</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{DANGO: Predicting Higher-Order Genetic Interactions}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Zhang, Ruochi and Bafna, Mihir and Ma, Jianzhu and Ma, Jian}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">jan</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2025}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{biorxiv}</span><span class="p">,</span>
  <span class="na">biorxiv</span> <span class="p">=</span> <span class="s">{10.1101/2020.11.26.400739v2}</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">{bioRxiv}</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.1101/2020.11.26.400739}</span><span class="p">,</span>
  <span class="na">elocation-id</span> <span class="p">=</span> <span class="s">{2020.11.26.400739}</span><span class="p">,</span>
  <span class="na">short</span> <span class="p">=</span> <span class="s">{biorxiv}</span><span class="p">,</span>
  <span class="na">shorttitle</span> <span class="p">=</span> <span class="s">{biorxiv}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> </ol> <h2 class="year">2023</h2> <ol class="bibliography"> <li> <div class="row"> <div class="col-sm-2 abbr"> <abbr class="badge">MLSB</abbr> </div> <div id="DiffRNAFold" class="col-sm-9"> <div class="title">DiffRNAFold: Generating RNA Tertiary Structures with Latent Space Diffusion</div> <div class="author"> <em>Mihir Bafna</em>, Vikranth Keerthipati, <a href="https://www.linkedin.com/in/subhash-kanaparthi/" target="_blank" rel="noopener noreferrer">Subhash Kanaparthi</a>, and <a href="https://ruochiz.com/" target="_blank" rel="noopener noreferrer">Ruochi Zhang</a> </div> <div class="periodical"> NeurIPS Workshop on Machine Learning for Structural Biology Dec 2023 </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> </div> <div class="abstract hidden"> <p>Abstract RNA molecules provide an exciting frontier for novel therapeutics. Accurate determination of RNA structure could accelerate development of therapeutics through an improved understanding of function. However, the extremely large conformation space has kept the RNA 3D structure space largely unresolved. Using recent advances in generative modeling, we propose DiffRNAFold, a latent space diffusion model for RNA tertiary structure design. Our preliminary results suggest that DiffRNAFold generated molecules are similar in 3D space to true RNA molecules, providing an important first step towards accurate structure and function prediction in vivo.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"> <abbr class="badge">ISMB</abbr> </div> <div id="Clarify" class="col-sm-9"> <div class="title">CLARIFY: Cell-cell interaction and gene regulatory network refinement from spatially resolved transcriptomics</div> <div class="author"> <em>Mihir Bafna</em>, <a href="https://sites.cc.gatech.edu/grads/h/hli691/" target="_blank" rel="noopener noreferrer">Hechen Li</a>, and <a href="https://xiuweizhang.wordpress.com/" target="_blank" rel="noopener noreferrer">Xiuwei Zhang</a> </div> <div class="periodical"> ISMB and Bioinformatics Jun 2023 </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://academic.oup.com/bioinformatics/article/39/Supplement_1/i484/7210503" class="btn btn-sm z-depth-0" role="button" target="_blank" rel="noopener noreferrer">Paper</a> <a href="https://github.com/MihirBafna/clarify#readme" class="btn btn-sm z-depth-0" role="button" target="_blank" rel="noopener noreferrer">Code</a> <a href="https://www.cc.gatech.edu/news/award-winning-computer-models-propel-research-cellular-differentiation" class="btn btn-sm z-depth-0" role="button" target="_blank" rel="noopener noreferrer">Press</a> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> </div> <div class="abstract hidden"> <p>Motivation: Gene regulatory networks (GRNs) in a cell provide the tight feedback needed to synchronize cell actions. However, genes in a cell also take input from, and provide signals to other neighboring cells. These cell-cell interactions (CCIs) and the GRNs deeply influence each other. Many computational methods have been developed for GRN inference in cells. More recently, methods were proposed to infer CCIs using single cell gene expression data with or without cell spatial location information. However, in reality, the two processes do not exist in isolation and are subject to spatial constraints. Despite this rationale, no methods currently exist to infer GRNs and CCIs using the same model. Results: We propose CLARIFY, a tool that takes GRNs as input, uses them and spatially resolved gene expression data to infer CCIs, while simultaneously outputs refined cell-specific GRNs. CLARIFY uses a novel multi-level graph autoencoder, which mimics cellular networks at a higher level and cell-specific GRNs at a deeper level. We applied CLARIFY to two real spatial transcriptomic datasets, one using seqFISH and the other using MERFISH, and also tested on simulated datasets from scMultiSim. We compared the quality of predicted GRNs and CCIs with state-of-the-art baseline methods that inferred either only GRNs or only CCIs. The results show that CLARIFY consistently outperforms the baseline in terms of commonly used evaluation metrics. Our results point to the importance of co-inference of CCIs and GRNs and to the use of layered graph neural networks as an inference tool for biological networks.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">Clarify</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Bafna, Mihir and Li, Hechen and Zhang, Xiuwei}</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{CLARIFY: Cell-cell interaction and gene regulatory network refinement from spatially resolved transcriptomics}</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{31st Conference on Intelligent Systems for Molecular Biology}</span><span class="p">,</span>
  <span class="na">short</span> <span class="p">=</span> <span class="s">{ISMB}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2023}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{Bioinformatics}</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">{39}</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">{Supplement_1}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{i484-i493}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">jun</span><span class="p">,</span>
  <span class="na">press</span> <span class="p">=</span> <span class="s">{https://www.cc.gatech.edu/news/award-winning-computer-models-propel-research-cellular-differentiation}</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">{Oxford University Press}</span><span class="p">,</span>
  <span class="na">appear</span> <span class="p">=</span> <span class="s">{true}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> </ol> <h2 class="year">2022</h2> <ol class="bibliography"> <li> <div class="row"> <div class="col-sm-2 abbr"> <abbr class="badge">NSUR</abbr> </div> <div id="ST-CCI" class="col-sm-9"> <div class="title">Benchmarking and Refining Cell-Cell Interactions with Spatial Transcriptomics and Deep Learning</div> <div class="author"> <em>Mihir Bafna</em>, and <a href="https://xiuweizhang.wordpress.com/" target="_blank" rel="noopener noreferrer">Xiuwei Zhang</a> </div> <div class="periodical"> St. Jude National Symposium for Undergraduate Research Oct 2022 </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://drive.google.com/file/d/1nEwQOVA5OmmaOwH0CeHw16U0P2CclKKD/view?usp=sharing" class="btn btn-sm z-depth-0" role="button" target="_blank" rel="noopener noreferrer">Poster</a> </div> <div class="abstract hidden"> <p>The (mal-)functioning of human tissues can be attributed to genes that are active (expressed) or repressed relative to expectations. New genomic technologies allow measurements not only at single cell (sc) resolution, but also retain information on the spatial location of the cell. These Spatial Transcriptomics (ST) technologies could revolutionize human health. For example, they offer an unprecedented look at the tumor microenvironment, revealing the infiltrating immune cells and their interactions with their cancerous counterparts. Many computational methods now analyze the complex, high dimensional ST data for inferring these cell-cell interactions (CCIs). However, the ST community lacks a centralized ground truth to holistically evaluate these tools. Here, (a) we systematically benchmark existing methods and (b) suggest a deep learning method for refining ST-CCI prediction. We evaluated 7 methods, including CellPhoneDB and DeepLinc, on 10 simulated datasets at 5 noise levels as well as 2 real datasets generated using SeqFISH and MerFISH. CellPhoneDB achieved an average precision/recall of 0.79/0.75 respectively, with the recall reducing to 0.65 for certain datasets. DeepLinc only achieved 0.68 prediction accuracy on SeqFISH data after being trained with labeled data. Additionally, the ROC curves were surprisingly linear, suggesting that increasing true positive rate comes only with increasing false positive rate. All other methods resulted in similar performance and the same pitfalls: failing to properly utilize either the spatial information or downstream gene regulatory interactions (GRN), thus increasing the false positive interactions. Our work provides, for the first time, a curated data resource for future tool comparisons and a systematic analysis of the shortcomings of existing methods. Lastly, to address these drawbacks, we deployed a preliminary version of a subgraph neural network, where we represent each cell by a subgraph of its underlying GRN, obtaining lower-dimensional representations of each cell with GRN information embedded. These embeddings incorporate gene expression, spatial information, and GRN activity thereby allowing us to refine ST-CCI inference. This has myriad applications like revealing interactions between co-located immune cells and tumor cells, addressing the central biological problem of why certain tumors are “immunologically hot” and respond better to immuno-oncotherapy.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"> <abbr class="badge">ACM-BCB</abbr> </div> <div id="DeepVifi" class="col-sm-9"> <div class="title">DeepViFi: Detecting Oncoviral Infections in Cancer Genomes Using Transformers</div> <div class="author">Utkrisht Rajkumar, Sara Javadzadeh,  <em>Mihir Bafna</em>, Dongxia Wu, Rose Yu, Jingbo Shang, and Vineet Bafna </div> <div class="periodical"> Proceedings of the 13th ACM International Conference on Bioinformatics, Computational Biology and Health Informatics Oct 2022 </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://dl.acm.org/doi/pdf/10.1145/3535508.3545551" class="btn btn-sm z-depth-0" role="button" target="_blank" rel="noopener noreferrer">Paper</a> <a href="https://github.com/UCRajkumar/DeepViFi" class="btn btn-sm z-depth-0" role="button" target="_blank" rel="noopener noreferrer">Code</a> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> </div> <div class="abstract hidden"> <p>We consider the problem of identifying viral reads in human host genome data. We pose the problem as open-set classification as reads can originate from unknown sources such as bacterial and fungal genomes. Sequence-matching methods have low sensitivity in recognizing viral reads when the viral family is highly diverged. Hidden Markov models have higher sensitivity but require domain-specific training and are difficult to repurpose for identifying different viral families. Supervised learning methods can be trained with little domain-specific knowledge but have reduced sensitivity in open-set scenarios. We present DeepViFi, a transformer-based pipeline, to detect viral reads in short-read whole genome sequence data. At 90% precision, DeepViFi achieves 90% recall compared to 15% for other deep learning methods. DeepViFi provides a semi-supervised framework to learn representations of viral families without domain-specific knowledge, and rapidly and accurately identify target sequences in open-set settings.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">DeepVifi</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Rajkumar, Utkrisht and Javadzadeh, Sara and Bafna, Mihir and Wu, Dongxia and Yu, Rose and Shang, Jingbo and Bafna, Vineet}</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{DeepViFi: Detecting Oncoviral Infections in Cancer Genomes Using Transformers}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2022}</span><span class="p">,</span>
  <span class="na">isbn</span> <span class="p">=</span> <span class="s">{9781450393867}</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">{Association for Computing Machinery}</span><span class="p">,</span>
  <span class="na">address</span> <span class="p">=</span> <span class="s">{New York, NY, USA}</span><span class="p">,</span>
  <span class="na">url</span> <span class="p">=</span> <span class="s">{https://doi.org/10.1145/3535508.3545551}</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.1145/3535508.3545551}</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{Proceedings of the 13th ACM International Conference on Bioinformatics, Computational Biology and Health Informatics}</span><span class="p">,</span>
  <span class="na">shorttitle</span> <span class="p">=</span> <span class="s">{ACM-BCB}</span><span class="p">,</span>
  <span class="na">short</span> <span class="p">=</span> <span class="s">{ACM-BCB}</span><span class="p">,</span>
  <span class="na">articleno</span> <span class="p">=</span> <span class="s">{2}</span><span class="p">,</span>
  <span class="na">numpages</span> <span class="p">=</span> <span class="s">{8}</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">{neural networks, viral detection, natural language processing, open-set classification}</span><span class="p">,</span>
  <span class="na">location</span> <span class="p">=</span> <span class="s">{Northbrook, Illinois}</span><span class="p">,</span>
  <span class="na">series</span> <span class="p">=</span> <span class="s">{BCB '22}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> </ol> <h2 class="year">2021</h2> <ol class="bibliography"><li> <div class="row"> <div class="col-sm-2 abbr"> <abbr class="badge">Patent</abbr> </div> <div id="MetaDetect" class="col-sm-9"> <div class="title">Computer-implemented methods for quantitation of features of interest in whole slide imaging</div> <div class="author">Nam Nguyen, Lorena Mora-Blanco, Kristen Turner, Julie Weise, Jason Christiansen, and <em>Mihir Bafna</em> </div> <div class="periodical"> Provisional patent. PCT/US2021/022308 Mar 2021 </div> <div class="links"> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@patent</span><span class="p">{</span><span class="nl">MetaDetect</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Computer-implemented methods for quantitation of features of interest in
  whole slide imaging}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Nguyen, Nam and Mora-Blanco, Lorena and Turner, Kristen and Weise, Julie and Christiansen, Jason and Bafna, Mihir}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2021}</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">{022308}</span><span class="p">,</span>
  <span class="na">filing</span> <span class="p">=</span> <span class="s">{Provisional patent. PCT/US2021/022308}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">mar</span><span class="p">,</span>
  <span class="na">notes</span> <span class="p">=</span> <span class="s">{filed March 15, 2021}</span><span class="p">,</span>
  <span class="na">abs</span> <span class="p">=</span> <span class="s">{}</span><span class="p">,</span>
  <span class="na">short</span> <span class="p">=</span> <span class="s">{Patent}</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li></ol> </div> </article> </div> </div> <footer class="fixed-bottom"> <div class="container mt-0"> © Copyright 2026 Mihir Bafna. Last updated: January 18, 2026. </div> </footer> <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script> <script src="https://cdn.jsdelivr.net/npm/@popperjs/core@2.11.2/dist/umd/popper.min.js" integrity="sha256-l/1pMF/+J4TThfgARS6KwWrk/egwuVvhRzfLAMQ6Ds4=" crossorigin="anonymous"></script> <script src="https://cdn.jsdelivr.net/npm/bootstrap@4.6.1/dist/js/bootstrap.min.js" integrity="sha256-SyTu6CwrfOhaznYZPoolVw2rxoY7lKYKQvqbtqN93HI=" crossorigin="anonymous"></script> <script src="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/js/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/masonry-layout@4.2.2/dist/masonry.pkgd.min.js" integrity="sha256-Nn1q/fx0H7SNLZMQ5Hw5JLaTRZp0yILA/FRexe19VdI=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/imagesloaded@4/imagesloaded.pkgd.min.js"></script> <script defer src="/assets/js/masonry.js" type="text/javascript"></script> <script type="text/javascript">$(function(){$('[data-toggle="tooltip"]').tooltip()});</script> <script src="https://cdn.jsdelivr.net/npm/medium-zoom@1.0.6/dist/medium-zoom.min.js" integrity="sha256-EdPgYcPk/IIrw7FYeuJQexva49pVRZNmt3LculEr7zM=" crossorigin="anonymous"></script> <script src="/assets/js/zoom.js"></script> <script src="/assets/js/common.js"></script> <script type="text/javascript">window.MathJax={tex:{tags:"ams"}};</script> <script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.2.0/es5/tex-mml-chtml.js"></script> <script defer src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script> <script async src="https://www.googletagmanager.com/gtag/js?id="></script> <script>function gtag(){dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],gtag("js",new Date),gtag("config","");</script> </body> </html>